pyrsia_node/src/main.rs

```diff
@@ -176,25 +193,52 @@ async fn main() {
 
     // Kick it off
     loop {
-        tokio::select! {
-            line = stdin.next_line() => {
-                let line = line.unwrap().expect("stdin closed");
-                debug!("next line!");
-                match tx2.send(line).await {
-                    Ok(_) => debug!("line sent"),
-                    Err(_) => error!("failed to send stdin input")
-                }
-            }
-            event = swarm.select_next_some() => {
-                if let SwarmEvent::NewListenAddr { address, .. } = event {
+        let evt = {
+            tokio::select! {
+                line = stdin.next_line() => Some(EventType::Input(line.expect("can get line").expect("can read line from stdin"))),
+                message = rx.recv() => Some(EventType::Message(message.expect("message exists"))),
+               // response = rx.recv() => Some(EventType::Response(response.expect("response exists"))),
+                event = swarm.select_next_some() =>  {
+                    if let SwarmEvent::NewListenAddr { address, .. } = event {
                     info!("Listening on {:?}", address);
-                }
+                    }
+
+                    //SwarmEvent::Behaviour(e) => panic!("Unexpected event: {:?}", e),
+                    None
+                },
             }
-            Some(message) = rx.recv() => {
-                info!("New message: {}", message);
-                swarm.behaviour_mut().floodsub_mut().publish(floodsub_topic.clone(), message.as_bytes());
-                swarm.behaviour_mut().lookup_blob(message).unwrap();
+        };
+
+        if let Some(event) = evt {
+            match event {
+                EventType::Response(resp) => {
```

Comment:
Maybe 

```suggestion
                EventType::SwarmMessage(resp) => {
```

sometimes we get random messages

Reply:
so this "Response" is of "EventType" enum in order to handle messages responses from multiple peers...for random messages we are already catching them at the last line. Here we are doing 3 things : 1) handling messages from command line 2) handling messages coming from channels to trigger Kademlia events 3) handling  responses for floodsub(it needs work to handle responses from "all" peers or "selected" peers .....as we get more clarity we can implement what messages need broadcast and what not

Reply:
I think it's less clear to combine all three "events" into one type to then separate them again and handle them.
Maybe have create three functions which handle their own events?

I see now... this was the old "GET hash" being sent on the network. It seems you lost https://github.com/pyrsia/pyrsia/pull/111/files/5ae4d6a965cc7250f354ea33efacd87f3fe8c082#diff-2c5d9cbb9033cbd429ad804a903684afdac3c2f5cb96bd0c474b56f0793521fbL196, is that intentional?

Reply:
Nope its there https://github.com/pyrsia/pyrsia/pull/111/files#diff-2c5d9cbb9033cbd429ad804a903684afdac3c2f5cb96bd0c474b56f0793521fbR229 . Here we are handling events in different functions but before passing the handle we have to  distinguish if the event is response, or am input from us or message from channel and hence the Eventype, it just matches it

Reply:
Rust is scary. Okay